internal/testplanet/satellite.go

```diff
@@ -234,10 +391,87 @@ func (planet *Planet) newSatellites(count int) ([]*SatelliteSystem, error) {
 		planet.databases = append(planet.databases, db)
 		planet.databases = append(planet.databases, pointerDB)
 
-		log.Debug("id=" + peer.ID().String() + " addr=" + peer.Addr())
+		api, err := planet.newAPI(i, identity, db, pointerDB, config, versionInfo)
+		if err != nil {
+			return xs, err
+		}
+		log.Debug("id=" + peer.ID().String() + " addr=" + api.Addr())
 
-		system := SatelliteSystem{Peer: *peer}
-		xs = append(xs, &system)
+		system := createNewSystem(log, peer, api)
+		xs = append(xs, system)
 	}
 	return xs, nil
 }
+
+// createNewSystem makes a new Satellite System and exposes the same interface from
+// before we split out the API. In the short term this will help keep all the tests passing
+// without much modification needed. However long term, we probably want to rework this
+// so it represents how the satellite will run when it is made up of many prrocesses.
+func createNewSystem(log *zap.Logger, peer *satellite.Peer, api *satellite.API) *SatelliteSystem {
+	system := &SatelliteSystem{
+		Peer: peer,
+		API:  api,
+	}
+	system.Log = log
+	system.Identity = peer.Identity
+	system.DB = api.DB
+
+	system.Dialer = api.Dialer
+
+	system.Contact.Service = api.Contact.Service
+	system.Contact.Endpoint = api.Contact.Endpoint
+
+	system.Overlay.DB = api.Overlay.DB
+	system.Overlay.Service = api.Overlay.Service
+	system.Overlay.Inspector = api.Overlay.Inspector
+
+	system.Discovery.Service = peer.Discovery.Service
+
+	system.Metainfo.Database = api.Metainfo.Database
+	system.Metainfo.Service = peer.Metainfo.Service
+	system.Metainfo.Endpoint2 = api.Metainfo.Endpoint2
+	system.Metainfo.Loop = peer.Metainfo.Loop
+
+	system.Inspector.Endpoint = api.Inspector.Endpoint
+
+	system.Orders.Endpoint = api.Orders.Endpoint
+	system.Orders.Service = peer.Orders.Service
+
+	system.Repair.Checker = peer.Repair.Checker
+	system.Repair.Repairer = peer.Repair.Repairer
+	system.Repair.Inspector = api.Repair.Inspector
+
+	system.Audit.Queue = peer.Audit.Queue
+	system.Audit.Worker = peer.Audit.Worker
+	system.Audit.Chore = peer.Audit.Chore
+	system.Audit.Verifier = peer.Audit.Verifier
+	system.Audit.Reporter = peer.Audit.Reporter
+
+	system.GarbageCollection.Service = peer.GarbageCollection.Service
+
+	system.DBCleanup.Chore = peer.DBCleanup.Chore
+
+	system.Accounting.Tally = peer.Accounting.Tally
+	system.Accounting.Rollup = peer.Accounting.Rollup
+	system.Accounting.ProjectUsage = peer.Accounting.ProjectUsage
+
+	system.Marketing.Listener = api.Marketing.Listener
+	system.Marketing.Endpoint = api.Marketing.Endpoint
+
+	system.GracefulExit.Chore = peer.GracefulExit.Chore
+	return system
+}
+
+func (planet *Planet) newAPI(count int, identity *identity.FullIdentity, db satellite.DB, pointerDB metainfo.PointerDB, config satellite.Config, versionInfo version.Info) (*satellite.API, error) {
```

Comment:
Since planet is a full network of uplinks, satellites, and storage nodes, could we call this `newSatelliteAPI`?

Reply:
same reasoning, since its in `satellite.go` https://github.com/storj/storj/pull/3237#discussion_r334581956